src/ng/compile.js

```diff
@@ -2592,11 +2587,13 @@ function $CompileProvider($provide, $$sanitizeUriProvider) {
             break;
 
           case '=':
-            if (optional && !attrs[attrName]) {
-              return;
+            if (!hasOwnProperty.call(attrs, attrName)) {
```

Comment:
Why not just use `attrs.hasOwnProperty` (instead of `hasOwnProperty.call(attrs`) here and above ?
(It is being used below anyway ;))